### PR TITLE
Refine contact merging.

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

File: SignalServiceKit/src/Contacts/Contact.m [hunk 1]
```diff
@@ -345,16 +345,28 @@ - (CNContact *)buildCNContactMergedWithNewContact:(CNContact *)newCNContact
         mergedCNContact.familyName = newCNContact.familyName.ows_stripped;
         mergedCNContact.nameSuffix = newCNContact.nameSuffix.ows_stripped;
     }
-    
+
+    if (mergedCNContact.organizationName.ows_stripped.length < 1) {
+        mergedCNContact.organizationName = newCNContact.organizationName.ows_stripped;
```
Comment (charlesmchen-signal) [inline SignalServiceKit/src/Contacts/Contact.m]: Handle organization name.

File: SignalServiceKit/src/Contacts/Contact.m [hunk 2]
```diff
@@ -345,16 +345,28 @@ - (CNContact *)buildCNContactMergedWithNewContact:(CNContact *)newCNContact
...
+    }
+
     // Phone Numbers
-    NSSet<PhoneNumber *> *existingPhoneNumberSet = [NSSet setWithArray:self.parsedPhoneNumbers];
-    
+    NSMutableSet<PhoneNumber *> *existingPhoneNumberSet = [NSMutableSet setWithArray:self.parsedPhoneNumbers];
+    [existingPhoneNumberSet addObjectsFromArray:self.userTextPhoneNumbers];
```
Comment (charlesmchen-signal) [inline SignalServiceKit/src/Contacts/Contact.m]: Don't drop unparse-able phone numbers unless they match an existing unparsed phone number.

Phone parsing is hard and I suggest we should be permissive about preserving phone numbers we don't understand. WDYT?

Comment (michaelkirk-signal) [reply SignalServiceKit/src/Contacts/Contact.m]: That seems fine. 

PhoneNumberSet *was* a set of PhoneNumbers, now it looks like it's a set of PhoneNumber's and NSString's, can you remove the incorrect annotation? Or make it an NSObject.

Comment (charlesmchen-signal) [reply SignalServiceKit/src/Contacts/Contact.m]: Whoops, fixed.

**Pull request merged**